### ISSUE TITLE
Angular 8 : Change detection

### ISSUE THREAD

**Issue by JessFarmerVA** [moribvndvs/ng2-idle#140]
**Describe the bug**
Change detection is not running onIdleEnd.

**To Reproduce**
```
    idle.onIdleEnd.subscribe(() => {
      this.counter = 30;
      this.isIdle = false;
      this.status = 'Welcome back';
      this.reset();
    });

    idle.onTimeout.subscribe(() => {
      this.status = 'Signing out';
    });


    idle.onTimeoutWarning.subscribe(() => {
      this.isIdle = true;
      this.status = 'Signing out123';
      console.log('PREPARING TO SIGN OUT!!!   timer: ', this.counter--);
    });
```
Change detection runs for onTimeoutWarning, and onTimeout, but not on onIdleEnd. The DOM does not update with the new status variable.

**Expected behavior**
Change detection should run and the DOM should update when I change variables inside onIdleEnd.

**Please tell us about your environment**
Desktop: 
Browser: Chrome, IE, Edge
- Version: Latest
- Angular version: 8.2.14
- NodeJS version:  11
- Language (TypeScript x.x, ES5, ES2015, etc.): Typescript 3.5.3

**Additional context**
I looked at the code, my thoughts are it may have something to do with zone.run()

**Comment by wingerlang** [comment]
I've found this bug as well, any plan to patch and fix it?

**Comment by Adelost** [comment]
Me too. @JessFarmerVA, did you find any good workaround?

One ugly workaround I found is to use setInterval to force change detection to occur:
E.g. by putting setInterval(() => {}, 1000) in the constructor of your root component. 

But doing this of course causes the change detection to run many times unnecessarily, and could hamper overall performance depending on what intervall value you use.

**Comment by JessFarmerVA** [comment]
@Adelost 
I ended up not using onIdleEnd and only used onTimeoutWarning and onTimeout... I managed to get my desired functionality with just those two..

However, I do remember finding a solution that involved importing something that could force change detection, not using setInterval. I believe it was tick() or something. I remember there was a few different way to force change Detection, and the one that worked was kind of ugly looking. Unfortunately I do not remember what it was though, sorry.
